sdk/all/src/main/java/io/opentelemetry/sdk/OpenTelemetrySdk.java

```diff
@@ -135,17 +145,39 @@ public Builder toBuilder() {
 
     private TracerProvider tracerProvider;
     private MeterProvider meterProvider;
+    private TracerSdkManagement tracerSdkManagement;
 
     /**
      * Sets the {@link TracerProvider} to use. This can be used to configure tracing settings by
-     * returning the instance created by a {@link TracerSdkProvider.Builder}.
+     * returning the instance created by a {@link TracerSdkProvider.Builder}. If the TracerProvider
```

Comment:
I think this is very confusing now. Why may `TracerProvider` want to implement `TracerSdkManagement`? The fact that `TracerSdkProvider` implements both of these interfaces is accidental(?) implementation detail.

And this still forces any user who wants to implement their own `TraceProvider` to think about `TracerSdkManagement`, which is not obvious. Btw, do we have any documentation/manual/example for implementing `TraceProvider`?

Reply:
I disagree, actually. What this requires is this: if you want to build a valid *`OpenTelemetrySdk`* (not `OpenTelemetry` in general), you have to provide something that implements the management interface. 

Every implementation of `OpenTelemetry` will have its own specific needs and requirements. The default SDK requires a management interface to be provided.

Reply:
Ok, but then if I want to change one of the `TracerProvider`/`TracerSdkManagement` I actually have to change both of them, correct? I cannot just call `setTracerProvider`, I have to call `setTracerSdkManagement`. So probably we should not have them both as separate methods? Instead have one method to set both.

Reply:
If your TracerProvider happens to implement both, you can set them both with just the one method, but I do agree things are pretty messy right now. Other solutions are welcome!